### PR TITLE
OCM-2544 | feat: support dns domain creation in rosa cli

### PR DESCRIPTION
**Pull request by marcolan018**
https://issues.redhat.com/browse/OCM-2544
Signed-off-by: marcolan018 <llan@redhat.com>

### CODE REVIEW

File: cmd/create/dnsdomains/cmd.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	defer r.Cleanup()
+
+	dnsdomain, err := r.OCMClient.CreateDNSDomain()
+	if err != nil {
+		r.Reporter.Errorf("Failed to create dns domain: %s", err)
+		os.Exit(1)
+	}
+
+	r.Reporter.Infof("DNS domain %s has been created.", dnsdomain.ID())
+	r.Reporter.Infof("To view all ingresses, run 'rosa list dns-domains")
```
Comment (oriAdler) [inline cmd/create/dnsdomains/cmd.go:48]:
```suggestion
	r.Reporter.Infof("To view all DNS domains, run 'rosa list dns-domains")
```

Comment (marcolan018) [reply cmd/create/dnsdomains/cmd.go]: fixed.
@oriAdler please help to review the changes again, thanks

File: cmd/create/dnsdomains/cmd.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	r := rosa.NewRuntime().WithOCM()
+	defer r.Cleanup()
+
+	dnsdomain, err := r.OCMClient.CreateDNSDomain()
+	if err != nil {
+		r.Reporter.Errorf("Failed to create dns domain: %s", err)
+		os.Exit(1)
+	}
+
+	r.Reporter.Infof("DNS domain %s has been created.", dnsdomain.ID())
```
Comment (oriAdler) [inline cmd/create/dnsdomains/cmd.go:47]:
```suggestion
	r.Reporter.Infof("DNS domain '%s' has been created.", dnsdomain.ID())
```

**Review by oriAdler — COMMENTED** [review]
Left a small comment, other than that, LGTM.

File: cmd/create/dnsdomains/cmd.go (hunk 2 above)
Comment (marcolan018) [reply cmd/create/dnsdomains/cmd.go]: @oriAdler 
thanks for the review, please check again, thanks

**Pull request merged**